crates/egui/src/widgets/image.rs

```diff
@@ -288,8 +288,24 @@ impl<'a> Image<'a> {
     }
 
     #[inline]
-    pub fn source(&self) -> &ImageSource<'a> {
-        &self.source
+    pub fn source(&'a self, ctx: &Context) -> ImageSource<'a> {
+        match &self.source {
+            ImageSource::Bytes { uri, .. } => match uri.starts_with("gif://") {
```

Comment:
Instead of using a special `gif://` URI, you can check for the GIF magic header, i.e. 

```rs
let is_gif = bytes.starts_with(b"GIF87a") || bytes.starts_with(b"GIF89a");
```

That means a user can use `include_image!` for gifs too, with no need for the special `include_gif!` macro